dccs/dccs_preprocessing_v1.R

```diff
@@ -0,0 +1,410 @@
+
+    nextCueAcc <- c(testDatTrim$key_resp_13.corr[2:(length(testDatTrim$key_resp_13.corr))], 2)
+    testDatTrim$nextCueAcc <- nextCueAcc
+
+    #create new vector identifying if current cue is same as prior, TRUE when cues differ and FALSE when cue is same as previous
+    #create new vector identifying if either switch and preswitch trial were answered erroneously, TRUE if either switch or preswitch was error
+    #remove first two trials
+    testDatTrim$switchStatus <- testDatTrim$cue != testDatTrim$priorCue
+    testDatTrim$toShapeSwitchPairErr <- (testDatTrim$cue == "COLOR" & testDatTrim$nextCue == "SHAPE" & (!testDatTrim$key_resp_13.corr | !testDatTrim$nextCueAcc)) |
+                                        (testDatTrim$cue == "SHAPE" & (!testDatTrim$key_resp_13.corr | !testDatTrim$priorCueAcc))
+    testDatTrim$toColorSwitchPairErr <- (testDatTrim$cue == "SHAPE" & (!testDatTrim$key_resp_13.corr | !testDatTrim$nextCueAcc)) |
+                                        (testDatTrim$cue == "COLOR" & testDatTrim$priorCue == "SHAPE" & (!testDatTrim$key_resp_13.corr | !testDatTrim$priorCueAcc))
+    testDatTrim <- testDatTrim[3:nrow(testDatTrim),]
+
+
+    #ANYSWITCH
+    #switch defined as above, comparison between switch group and nonswitch group
+
+    #subset the data for switch versus nonswitch trials, create new dataframes for each
+    testDatTrim_anySwitch <- testDatTrim[testDatTrim$switchStatus, ]
+    testDatTrim_anySwitchOPP <- testDatTrim[!testDatTrim$switchStatus, ]
+
+    #compute mean accuracy and standard error for switch versus nonswitch trials
+    anySwitch_meanACC <- mean(testDatTrim_anySwitch$key_resp_13.corr)
+    anySwitchOPP_meanACC <- mean(testDatTrim_anySwitchOPP$key_resp_13.corr)
+
+    anySwitch_seACC <- sd(testDatTrim_anySwitch$key_resp_13.corr)/sqrt(length(testDatTrim_anySwitch$key_resp_13.corr))
+    anySwitchOPP_seACC <- sd(testDatTrim_anySwitchOPP$key_resp_13.corr)/sqrt(length(testDatTrim_anySwitchOPP$key_resp_13.corr))
+
+    #subset the data for correct trials only, separately for switch versus nonswitch trials, creating new dataframes for each
+    testDatTrim_anySwitchCorr <- testDatTrim_anySwitch[testDatTrim_anySwitch$key_resp_13.corr==1, ]
+    testDatTrim_anySwitchOPPCorr <- testDatTrim_anySwitchOPP[testDatTrim_anySwitchOPP$key_resp_13.corr==1, ]
+
+    #for correct trials, for both switch and nonswitch groups, compute mean RT (raw and log-corrected), standard error, and median RT
+    anySwitchCorr_meanRT <- mean(testDatTrim_anySwitchCorr$key_resp_13.rt)
+    anySwitchOPPCorr_meanRT <- mean(testDatTrim_anySwitchOPPCorr$key_resp_13.rt)
+
+    anySwitchCorr_seRT <- sd(testDatTrim_anySwitchCorr$key_resp_13.rt)/sqrt(length(testDatTrim_anySwitchCorr$key_resp_13.rt))
+    anySwitchOPPCorr_seRT <- sd(testDatTrim_anySwitchOPPCorr$key_resp_13.rt)/sqrt(length(testDatTrim_anySwitchOPPCorr$key_resp_13.rt))
+
+    anySwitchCorr_logMeanRT <- mean(log((1+testDatTrim_anySwitchCorr$key_resp_13.rt)))
```

Comment:
I was wondering if there is a reason for double "((" after log or is it a typo?
I am sure it is insignificant and make no difference if it a typo, but I was curious about it

Reply:
Looks wholly unnecessary!  Oops!